clang/tools/libclang/CCAS.cpp

```diff
@@ -492,9 +492,11 @@ CXCASReplayResult clang_experimental_cas_replayCompilation(
 
   SmallString<256> DiagText;
   std::optional<int> Ret;
+  bool UseCASBackend = Invok->getCodeGenOpts().UseCASBackend;
```

Comment:
It's better to move the check down in `CompileJobCache::replayCachedResult`. Remove the `UseCASBackend` parameter from the `CompileJobCache::replayCachedResult` function and have itself check the codegen option, so the callers don't have to do it.

Also add a test by producing an object file via a normal clang invocation, then an object file by replaying a cached compilation through libclang (using `test/CAS/libclang-replay-job.c` as reference) and making sure object file contents are the same using `diff`.

Reply:
Sounds good!